bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -172,7 +173,7 @@ public static BType getType(Object value) {
             return BTypes.typeInt;
         } else if (value instanceof Double) {
             return BTypes.typeFloat;
-        } else if (value instanceof DecimalValue) {
+        } else if (value instanceof DecimalValue || value instanceof BigDecimal) {
```

Comment:
decimals are stored as DecimalValue object which has the bigdecimal value. hence do we need this?

Reply:
fixed